controller/api/destination/external-workload/endpoints_reconciler_test.go

```diff
@@ -49,40 +54,87 @@ var (
 // === Test create / update / delete ===
 
 // Test that when a service has no endpointslices written to the API Server, reconciling
-// with a workload will create a new endpointslice.
-func TestReconcilerCreatesNewEndpointSlice(t *testing.T) {
+// with a workload will create new endpointslices (one per IP family)
+func TestReconcilerCreatesNewEndpointSlices(t *testing.T) {
 	// We do not need to receive anything through the informers so
 	// create a client with no cached resources
 	k8sAPI, err := k8s.NewFakeAPI([]string{}...)
 	if err != nil {
 		t.Fatalf("unexpected error when creating Kubernetes clientset: %v", err)
 	}
 
-	svc := makeIPv4Service(map[string]string{"app": "test"}, []corev1.ServicePort{httpUnnamedPort}, "")
-	ew := makeExternalWorkload("1", "wlkd-1", map[string]string{"app": ""}, map[int32]string{8080: ""}, []string{"192.0.2.0"})
-	ew.ObjectMeta.UID = types.UID(fmt.Sprintf("%s-%s", ew.Namespace, ew.Name))
+	for _, tc := range []struct {
```

Comment:
It seems there is no way to enforce that if we have two addresses, they should be of different type. Do you think its worth adding that and testing for it?

Reply:
We don't have a validating webhook for this resource, and I don't see a way of doing this with plain OpenAPI validation. Perhaps doable with CEL once we move to a MSKV 1.25?